Celeste.Mod.mm/Patches/MapData.cs

```diff
@@ -96,6 +96,8 @@ public patch_MapData(AreaKey area)
 
             } catch (Exception e) {
                 Mod.Logger.Log(LogLevel.Warn, "misc", $"Failed loading MapData {Area}");
+                if (e is OutOfMemoryException)
+                    throw e; // Unrecoverable
```

Comment:
The `throw;` syntax rethrows the caught exception without altering its stack trace.

Better yet, it should be possible to use [`when` to let the runtime filter the exception before catching it.](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/when)